### DISCUSSION TITLE
SIGHUP handler

### DISCUSSION THREAD

**Discussion by HiHat** [Waujito/youtubeUnblock#171] [Ideas]
Большая просьба добавить обработчик сигнала HUP (игнорирование), т.к. при запуске вручную и завершении терминальной сессии процесс убивается системой!  Правила iptables остаются, но вроде не мешают.
При запуске через nohup продолжает работать!

**Comment by Waujito** [comment]
Тут скорее надо не sighup добавлять, а daemonize флаг, чтобы отключать от терминала. С этим сигхапом может возникнуть много проблем, когда юзер ожидает от анблока завершения, а он сам демонизируется.

**Reply by HiHat** [reply]
Флаг - вообще круто,  но SIGHUP обычно игнорируют в демонах, иначе бесполезно его вместе с & запускать...

**Reply by Waujito** [reply]
Если у меня будет открыт баш шелл через ssh на роутере, а я возьму и вырву кабель интернета, оно же сигхап отправит?

**Reply by HiHat** [reply]
Если кабель LAN порвать, то, по идее, сессия должна будет порваться и всем процессам в сессии придёт SIGHUP.

**Reply by HiHat** [reply]
Я добавил  nohup в init скрипт, чтобы все продолжало работать после ручного завершения терминального сеанса

**Reply by Waujito** [reply]
Добавил флаг --daemonize, а также --syslog для логирования в системный лог, а не в stdout

**Reply by HiHat** [reply]
Огромное спасибо!
Большая просьба выпустить релиз, а то многие могут не понять, что актуальные рабочие версии надо из actions брать!
